src/inputs/checkbox.scss

```diff
@@ -31,7 +31,7 @@
     // Adds hover to everything that is not disabled
     &:hover > input:enabled ~ .iui-checkbox-checkmark {
       @include themed {
```

Comment:
So for disabled checkmark, background is still there?

Reply:
This is only for hover on enabled visibility checkboxes.  All disabled checkboxes have the same background color (hover or no hover) across all checkboxes.

Reply:
Then it won't really solve this issue.
![image](https://user-images.githubusercontent.com/9084735/143049528-f9ebe60d-2808-4d47-82f6-c270490354c5.png)

Unless that is the intended look

Reply:
I believe Vilius' screenshot is when hovered.  Here is a screenshot of this PR's checkboxes with a pink background.  Top checkbox is hovered (allows some of the pink bg to come thru due to opacity) and bottom checkbox has no hover.
![Screen Shot 2021-11-23 at 10 22 37 AM](https://user-images.githubusercontent.com/849817/143052219-d7ff39af-5c63-4b53-934a-1ef16a8a1e51.png)